### PR TITLE
Maps all unique content-types for operations and set passthroughBehavior to never in CORS preflight integration

### PR DESCRIPTION
**Pull request by jfkisafk**
#### Background
* What do these changes do? 
  * Maps all unique content-types for operations in CORS preflight integration template, in addition to the `application/json` template that is currently added.
  * Sets the `passthroughBehavior` for the mock integration to "never" to fail-fast with _415 Unsupported_ in case a request is made with unsupported Content-Type.
  * Both these features are behind a new OpenAPI configuration `syncCorsPreflightIntegration` (not sure about the name), which the service teams can turn on.
* Why are they important?
  * So that API Gateway does not error out when OPTIONS request is made with a custom payload and a Content-Type other than `application/json`. This will pass the request to the actual GET/PUT/... method.
  * Requests made with unsupported Content-Type fail fast with _415 Unsupported_ instead of receiving _500 InternalServerError_ from API Gateway.

#### Testing
* How did you test these changes?
  - [x] Added test cases for the new changes.
  - [x] Checked backwards compatibility by running `./gradlew clean build`.

#### Links
* closes smithy-lang/smithy#2289
* https://docs.aws.amazon.com/apigateway/latest/developerguide/integration-passthrough-behaviors.html

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jfkisafk** [comment]
Please lmk if this feature should be launched behind a feature flag in openapi settings or if I need to update the docs as well?

File: smithy-aws-apigateway-openapi/src/main/java/software/amazon/smithy/aws/apigateway/openapi/AddCorsPreflightIntegration.java [hunk 1]
```diff
@@ -214,21 +217,41 @@ private static ResponseObject createPreflightResponse(Map<CorsHeader, String> he
...
         for (Map.Entry<CorsHeader, String> e : headers.entrySet()) {
             responseBuilder.putResponseParameter("method.response.header." + e.getKey(), "'" + e.getValue() + "'");
         }
 
+        boolean isPreflightSynced = Boolean.TRUE.equals(context.getConfig().getSyncCorsPreflightIntegration());
         MockIntegrationTrait.Builder integration = MockIntegrationTrait.builder()
                 // See https://forums.aws.amazon.com/thread.jspa?threadID=256140
                 .contentHandling("CONVERT_TO_TEXT")
-                .passThroughBehavior("when_no_match")
+                .passThroughBehavior(isPreflightSynced ? "never" : "when_no_match")
```
Comment (kstich) [inline smithy-aws-apigateway-openapi/src/main/java/software/amazon/smithy/aws/apigateway/openapi/AddCorsPreflightIntegration.java:233]: Can there be a link to [the `passThroughBehavior` docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/integration-passthrough-behaviors.html) here?

**Review by kstich — CHANGES_REQUESTED** [review]

File: docs/source-2.0/guides/model-translations/converting-to-openapi.rst [hunk 2]
```diff
@@ -691,6 +691,29 @@ onErrorStatusConflict (``String``)
...
+            "version": "1.0",
+            "plugins": {
+                "openapi": {
+                    "service": "example.weather#Weather",
+                    "syncCorsPreflightIntegration": true
+                }
+            }
+        }
+
+    With this enabled, the ``passthroughBehavior`` for the CORS preflight integration
```
Comment (kstich) [inline docs/source-2.0/guides/model-translations/converting-to-openapi.rst:713]: Can there be a link to [the passThroughBehavior docs](https://docs.aws.amazon.com/apigateway/latest/developerguide/integration-passthrough-behaviors.html) here?

File: docs/source-2.0/guides/model-translations/converting-to-openapi.rst [hunk 3]
```diff
@@ -691,6 +691,29 @@ onErrorStatusConflict (``String``)
             }
         }
 
+.. _generate-openapi-setting-syncCorsPreflightIntegration:
```
Comment (kstich) [inline docs/source-2.0/guides/model-translations/converting-to-openapi.rst:694]: This setting is API-Gateway specific, so should be moved down to [the API Gateway settings section](https://github.com/smithy-lang/smithy/blob/main/docs/source-2.0/guides/model-translations/converting-to-openapi.rst?plain=1#L1450). It's worth also adding a note to [the API Gateway specific CORS section](https://github.com/smithy-lang/smithy/blob/main/docs/source-2.0/guides/model-translations/converting-to-openapi.rst?plain=1#L1645).

Comment (kstich) [reply docs/source-2.0/guides/model-translations/converting-to-openapi.rst:694]:
```suggestion
.. _generate-openapi-apigateway-setting-syncCorsPreflightIntegration:
```

**Review by kstich — APPROVED** [review]

**Pull request merged**